test/e2e/e2e_test.go

```diff
@@ -177,6 +178,145 @@ func TestLowNodeUtilization(t *testing.T) {
+	}
+
+	nodes, err := nodeutil.ReadyNodes(ctx, clientset, nodeInformer, "", nil)
+	if err != nil {
+		klog.Fatalf("%v", err)
+	}
+
+	maxPodLifeTimeSeconds := uint(1)
+	strategies.PodLifeTime(
+		ctx,
+		clientset,
+		deschedulerapi.DeschedulerStrategy{
+			Enabled: true,
+			Params: &deschedulerapi.StrategyParameters{
+				MaxPodLifeTimeSeconds: &maxPodLifeTimeSeconds,
+				Namespaces:            namespaces,
+			},
+		},
+		nodes,
+		evictions.NewPodEvictor(
+			clientset,
+			evictionPolicyGroupVersion,
+			false,
+			0,
+			nodes,
+			false,
+		),
+	)
+	time.Sleep(10 * time.Second)
+}
+
+func getPodNames(pods []v1.Pod) []string {
+	names := []string{}
+	for _, pod := range pods {
+		names = append(names, pod.Name)
+	}
+	return names
+}
+
+func TestNamespaceConstraints(t *testing.T) {
+	ctx := context.Background()
+
+	clientSet, err := client.CreateClient(os.Getenv("KUBECONFIG"))
+	if err != nil {
+		t.Errorf("Error during client creation with %v", err)
+	}
+
+	stopChannel := make(chan struct{}, 0)
+
+	sharedInformerFactory := informers.NewSharedInformerFactory(clientSet, 0)
+	sharedInformerFactory.Start(stopChannel)
+	sharedInformerFactory.WaitForCacheSync(stopChannel)
+	defer close(stopChannel)
+
+	nodeInformer := sharedInformerFactory.Core().V1().Nodes()
+
+	testNamespace := &v1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: "e2e-" + strings.ToLower(t.Name())}}
+	if _, err := clientSet.CoreV1().Namespaces().Create(ctx, testNamespace, metav1.CreateOptions{}); err != nil {
+		t.Fatalf("Unable to create ns %v", testNamespace.Name)
+	}
+	defer clientSet.CoreV1().Namespaces().Delete(ctx, testNamespace.Name, metav1.DeleteOptions{})
+
+	rc := RcByNameContainer("test-rc-toomanyrestarts", testNamespace.Name, 5, map[string]string{"test": "toomanyrestarts"}, nil)
+	if _, err := clientSet.CoreV1().ReplicationControllers(rc.Namespace).Create(ctx, rc, metav1.CreateOptions{}); err != nil {
+		t.Errorf("Error creating deployment %v", err)
+	}
+	defer deleteRC(ctx, t, clientSet, rc)
+
+	// wait for a while so all the pods are at least few seconds older
+	time.Sleep(5 * time.Second)
+
+	// it's assumed all new pods are named differently from currently running -> no name collision
+	podList, err := clientSet.CoreV1().Pods(rc.Namespace).List(ctx, metav1.ListOptions{LabelSelector: labels.SelectorFromSet(rc.Spec.Template.Labels).String()})
+	if err != nil {
+		t.Fatalf("Unable to list pods")
+	}
+
+	if len(podList.Items) != 5 {
+		t.Fatalf("Expected 5 replicas, got %v instead", len(podList.Items))
+	}
+
+	podNamesInclude := make(map[string]struct{})
+	for _, pod := range podList.Items {
+		podNamesInclude[pod.Name] = struct{}{}
+	}
+
+	t.Logf("Existing pods: %v", getPodNames(podList.Items))
+	t.Logf("set the strategy to delete pods from %v namespace", rc.Namespace)
+	runPodLifetimeStrategy(ctx, clientSet, nodeInformer, deschedulerapi.Namespaces{
+		Include: []string{rc.Namespace},
+	})
+
+	t.Logf("Waiting 10s")
+	time.Sleep(10 * time.Second)
```

Comment:
duplicate `time.Sleep` (you also call it at the end of `runPodLifetimeStrategy`)

btw `Sleep`s can introduce flakes if for whatever reason the test doesn't run as smoothly as you assume

Reply:
I removed the one from `runPodLifetimeStrategy`. Also, replaced the first Sleep(10) with wait.Pool since we are waiting until all pods get deleted.